Add support for manifest variable substitution

ioben opened the pull request:
Adds two new parameters for `out`: `vars` and `vars_files`.
The `vars_files` are passed directly to the `cf push` command through the `--vars-file` argument.
The `vars` are dumped as yaml to the current working directory, and the file is appended to the array of `vars_files`.

Needed to bump CF CLI version to the current latest to support variable substitution.

This is a bit of a first pass, definitely would like feedback. It has already proved useful for me though.

Maxime-Lafond commented:
What happen with this PR ? Can we proceed to the merge ?

ukabu commented:
Please :
- [ ] remove the cf cli update from this PR. and rebase from `master`
- [ ] declare `args` using `const`. `args` should also have a more explicit name.
- [ ] replace `if (x) args = args.concat("x", x)` with `x && args.push("x", x)`
- [ ] use arrow function instead of function
- [ ] provide a test script
- [ ] use `const` instead of `let`

ioben commented:
@ukabu 

1. Done
2. Renamed to `cf_args`. Also moved from using `concat` (non-mutating) to `push` (mutating). Added `let` instead of `const` to signal the array changes.
3. Done
4. Fixed. Shows the last time I wrote Javascript 😨
5. I struggled a bit with testing. I looked in the `package.json` which has a test script defined to run `jest`, but I don't think this project actually uses jest? I just manually ran each of the shell scripts in the `tests` directory, let me know if there's a better way to handle this.
  I've added three tests for `vars` and `var_files`. I was able to get every other test to pass, except `./test-docker-static-in.sh`. I made a few improvements while going through the tests, I can make them a separate PR.
6. Done for `out.js`'s `vars_files`.

ioben commented:
Any update on this? I have two other PRs I'd love to submit, but they're waiting on this one

ukabu commented:
very busy. will take a look and merge at the latest early next week.

ukabu commented:
Merged! Thanks for your patience.
I've renamed the out params to manifest_vars and manifest_vars_files to be a bit clearer about the usage of the params.
close #1
